### PR TITLE
repart: Allow Subvolumes= when running offline

### PR DESCRIPTION
**Pull request by daandemeyer**
mkfs.btrfs will soon learn a new option --subvol so let's stop failing
when Subvolumes= is used offline and use the new --subvol option instead
to create subvolumes in the generated root filesystem without root privileges
or loop devices.

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
Patch isn't merged yet (or even public yet), but I tested it already and it works so putting up the PR already. Will undraft once the option has been merged into mkfs.btrfs

**Comment by DaanDeMeyer** [comment]
Patch: https://lore.kernel.org/linux-btrfs/20240627095455.315620-1-maharmstone@fb.com/T/#u

File: src/partition/repart.c [hunk 1]
```diff
@@ -5153,6 +5149,29 @@ static int partition_populate_filesystem(Context *context, Partition *p, const c
         return 0;
 }
 
+static int finalize_extra_mkfs_options(const Partition *p, const char *root, char ***ret) {
+        int r;
+        char **sv = NULL;
+
+        assert(ret);
```
Comment (bluca) [inline src/partition/repart.c:5156]: assert(p)

**Review by bluca — APPROVED** [review]

File: src/partition/repart.c [hunk 2]
```diff
@@ -5153,6 +5148,40 @@ static int partition_populate_filesystem(Context *context, Partition *p, const c
         return 0;
 }
 
+static int finalize_extra_mkfs_options(const Partition *p, const char *root, char ***ret) {
+        int r;
+        char **sv = NULL;
```
Comment (bluca) [inline src/partition/repart.c]: `_cleanup_strv_` ?

**Comment by DaanDeMeyer** [comment]
Back to draft since we're now waiting for https://github.com/kdave/btrfs-progs/pull/872 to go in

File: src/partition/repart.c [hunk 3]
```diff
@@ -5156,6 +5148,40 @@ static int partition_populate_filesystem(Context *context, Partition *p, const c
...
+
+        r = mkfs_options_from_env("REPART", p->format, &sv);
+        if (r < 0)
+                return log_error_errno(r,
+                                       "Failed to determine mkfs command line options for '%s': %m",
+                                       p->format);
+
+        if (partition_needs_populate(p) && root && streq(p->format, "btrfs")) {
+                STRV_FOREACH(subvol, p->subvolumes) {
+                        if (p->default_subvolume && streq(*subvol, p->default_subvolume))
```
Comment (poettering) [inline src/partition/repart.c:5166]: streq_ptr()?

**Pull request merged**